typegate/src/utils.ts

```diff
@@ -214,3 +225,82 @@ export function collectFieldNames(tg: TypeGraph, typeIdx: number) {
   }
   return { title: typ?.title, fields: [] };
 }
+
+/**
+ * base64 decode and untar at cwd/{dir}
+ */
+export async function uncompress(dir: string, tarb64: string) {
+  const baseDir = path.join(Deno.cwd(), dir);
+  const buffer = base64.decode(tarb64);
+  const streamReader = new Blob([buffer])
+    .stream()
+    .pipeThrough(new DecompressionStream("gzip"))
+    .getReader();
+  const denoReader = conversion.readerFromStreamReader(streamReader);
+
+  const untar = new Untar(denoReader);
+  const entries = [];
+  for await (const entry of untar) {
+    entries.push(entry.fileName);
+    if (entry.fileName == ".") {
+      continue;
+    }
+    if (entry.type === "directory") {
+      const resDirPath = path.join(baseDir, entry.fileName);
+      await ensureDir(resDirPath);
+      continue;
+    }
+    const resFilePath = path.join(baseDir, entry.fileName);
+    await ensureFile(resFilePath);
+    const file = await Deno.open(resFilePath, { write: true });
+    await conversion.copy(entry, file);
+    file.close();
+  }
+  return baseDir;
+}
+
+export async function digestPath(path: string) {
+  const data = new TextEncoder().encode(path);
+  const buffer = await crypto.subtle.digest("SHA-1", data);
```

Comment:
there should be a hash function inside `crypto.ts` you can reuse

Reply:
It encodes to base64 (has special characters)